dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -90,27 +93,89 @@ HRESULT CALLBACK DrivesContextMenuCallback(IShellFolder *psf,
 
         if (GetVolumeInformationA(szDrive, NULL, 0, NULL, NULL, &dwFlags, NULL, 0))
         {
-            /* Disable format if read only */
-            if (!(dwFlags & FILE_READ_ONLY_VOLUME) && GetDriveTypeA(szDrive) != DRIVE_REMOTE)
+            if (!(dwFlags & FILE_READ_ONLY_VOLUME) && nDriveType != DRIVE_REMOTE)
             {
+                /* add separator and Format */
                 _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, 0, MFT_SEPARATOR, NULL, 0);
                 _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, pqcminfo->idCmdFirst++, MFT_STRING, MAKEINTRESOURCEW(IDS_FORMATDRIVE), MFS_ENABLED);
             }
+            if (nDriveType == DRIVE_CDROM || nDriveType == DRIVE_REMOTE)
```

Comment:
Ejection makes no sense for a remote drive. It should be reserved to CDROM/FLOPPY.
For remote drive, you should add an item "Disconnect" that would rely on MPR to perform the removal.

Reply:
OK. The drive targets are DRIVE_REMOVABLE, DRIVE_CDROM, and DRIVE_REMOTE.